src/lib.rs

```diff
@@ -380,15 +390,18 @@ impl RecvFlags {
 /// A version of [`IoSliceMut`] that allows the buffer to be uninitialised.
 ///
 /// [`IoSliceMut`]: std::io::IoSliceMut
+#[cfg(not(target_os = "wasi"))]
 #[repr(transparent)]
 pub struct MaybeUninitSlice<'a>(sys::MaybeUninitSlice<'a>);
```

Comment:
It seems like this PR would be a lot simpler if this type also exists on wasi. Or, alternatively, if the type was moved to a module so that we needed only a single `#[cfg]` rather than 11 of them.